src/configurationProvider.ts

```diff
@@ -134,7 +134,9 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                     }).sort ((a, b): number => {
                         return a.label > b.label ? 1 : -1;
                     });
-                    const selection = await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" });
+                    const selection = pickItems.length > 1 ?
+                        await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" })

```

Comment:
Select main class<project name> ?? Not understand which should be selected? Both?

Reply:
yes, if it is not specified in launch.json, java debug will list them for the choosing from user, if only one main class is found, I will use it directly without popup windows for one chosen option.